pkg/systemd/services/service.jsx

```diff
@@ -21,50 +21,130 @@ import React from "react";
 import { Breadcrumb, BreadcrumbItem } from "@patternfly/react-core/dist/esm/components/Breadcrumb/index.js";
 import { Page, PageSection } from "@patternfly/react-core/dist/esm/components/Page/index.js";
 import { Gallery, GalleryItem } from "@patternfly/react-core/dist/esm/layouts/Gallery/index.js";
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
 
+import { EmptyStatePanel } from "cockpit-components-empty-state.jsx";
 import { ServiceDetails } from "./service-details.jsx";
 import { LogsPanel } from "cockpit-components-logs-panel.jsx";
 import { superuser } from 'superuser';
 import { WithDialogs } from "dialogs.jsx";
 
+import s_bus from "./busnames.js";
+
 import cockpit from "cockpit";
 
 const _ = cockpit.gettext;
 
+function debug() {
+    if (window.debugging == "all" || window.debugging?.includes("service-details"))
+        console.debug.apply(console, arguments);
+}
+
 export class Service extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            /* The initial load of the Services page will not call GetAll for units Properties
-             * since ListUnits API call already has provided us with a subset of the Properties.
-             * As a result, properties like the 'Requires' are not present in the state at this point.
-             * If it's the first time to open this service's details page we need to fetch
-             * the unit properties by calling getUnitByPath.
-             */
-            shouldFetchProps: props.unit.Names === undefined,
+            unit_id: null,
+            unit_props: null,
+            error: null,
+            reloading: false,
         };
+
+        this.path = null;
+        this.subscriptions = [];
+    }
+
+    async componentDidMount() {
+        const dbus = this.props.dbusClient;
+        const [path] = await dbus.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [this.props.unitId]);
+        this.path = path;
+
+        this.subscriptions.push(dbus.subscribe(
```

Comment:
Do we need to worry that `componentWillUnmount` runs before `componentDidMount` is done?

Reply:
https://reactjs.org/docs/react-component.html#componentdidmount does not explicitly mention it. It sounds a bit weird to expect React to call WillUnmount before it called DidMount, and I haven't observed that behaviour. But it would be safe even -- `this.subscriptions` is initialized to `[]`, not to `null`, so it wouldn't crash.

Reply:
No, I meant that although componentDidMount is marked as async, I don't think that React will await it, correct?  So from the point of view of React, componentDidMount is done when reaches the `await dbus.call` statement, and it might call componentWillUnmount at any time, before the rest of the code in componentDidMount has executed.

I think it's gonna be fine.